File: back/api/Models/Product/Question/ProductQuestion.cs
﻿using System.ComponentModel.DataAnnotations;

namespace api.Models.Product.Question;

public class ProductQuestion
{
  [Key] public Guid Id { get; init; }
  [Required] public int ProductId { get; init; }
  [Required] public string QuestionersNickname { get; init; }
  [Required] public string Title { get; init; }
  [Required] public string Content { get; init; }
  [Required] public DateTimeOffset CreatedAt { get; init; }
  [Required] public bool IsApproved { get; set; }

  public virtual IEnumerable<ProductQuestionAnswer> Answers { get; init; }
}

File: front/types/ProductQuestion.ts
import { Product } from "./Product";

export type ProductQuestionAnswer = {
  id: string;
  questionId: string;
  answerersNickname: string;
  content: string;
  createdAt: string;
  byEmployee: boolean;
};

export type ProductQuestion = {
  id: string;
  productId: number;
  questionersNickname: string;
  title: string;
  content: string;
  createdAt: string;
  answers: ProductQuestionAnswer[];
};

export type NotApprovedProductQuestion = ProductQuestion & {
  product: Product;
};

export type AddProductQuestionRequestBody = Omit<
  ProductQuestion,
  "id" | "reviewId" | "byEmployee" | "createdAt" | "answers" | "productId"
>;
